### PR TITLE
Use new prometheus deployment in test framework

### PR DESCRIPTION
**Pull request by howardjohn**
This switches to using the new samples/addons prometheus deployment for our testing. The mixer tests are still using the old istioctl tests, as is the security test which validates prometheus certs, so we still have coverage there

For https://github.com/istio/istio/issues/23583

### CODE REVIEW

File: manifests/addons/values-prometheus.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ server:
   # Use port 9090 to match Istio documentation
   service:
     servicePort: 9090
-
+  readinessProbeInitialDelay: 0
```
Comment (howardjohn) [inline manifests/addons/values-prometheus.yaml:16]: The default is 30s which is super slow for tests (and users!). our current install uses 0 so this seems reasonable (and is k8s default)

File: tests/integration/mixer/outboundtrafficpolicy/traffic_test.go [hunk 2]
```diff
@@ -50,8 +50,8 @@ components:
   - enabled: true
   telemetry:
     enabled: true
-addonComponents:
-  prometheus:
+addonComponents:
```
Comment (dgn) [inline tests/integration/mixer/outboundtrafficpolicy/traffic_test.go:53]: nit: extra whitespace

**Review by dgn — APPROVED** [review]

**Review by mandarjog — APPROVED** [review]
Lgtm

**Review by elfinhe — APPROVED** [review]

**Review by incfly — APPROVED** [review]

**Pull request merged**